2024-11-14 11:34:48.629 | INFO     | sc2.protocol:_execute:72 - Client status changed to Status.launched (was None)
2024-11-14 11:34:48.630 | INFO     | sc2.controller:create_game:37 - Creating new game
2024-11-14 11:34:48.631 | INFO     | sc2.controller:create_game:38 - Map:     Equilibrium513AIE
2024-11-14 11:34:48.632 | INFO     | sc2.controller:create_game:39 - Players: Bot LadderBot(Terran), Computer VeryHard(Zerg, RandomBuild)
2024-11-14 11:34:48.635 | INFO     | sc2.protocol:_execute:72 - Client status changed to Status.init_game (was Status.launched)
2024-11-14 11:35:01.594 | INFO     | sc2.protocol:_execute:72 - Client status changed to Status.in_game (was None)
2024-11-14 11:35:01.596 | INFO     | sc2.main:_play_game:221 - Player 1 - Bot LadderBot(Terran)
2024-11-14 11:35:02.579 | 00:00    0ms    0M    0G  12/ 15U [Version] 5.0.13.92440
2024-11-14 11:35:02.580 | 00:00    0ms    0M    0G  12/ 15U [Version] 5.0.6 Balance mod recognized.
2024-11-14 11:35:02.727 | 00:00    0ms    0M    0G  12/ 15U [ZoneManager] Map set to: MapName.Unknown from name: Equilibrium AIE and hash: 3540971.
2024-11-14 11:35:04.319 | [DataManager] No BuildDetector found, enemy build data cannot be saved
2024-11-14 11:35:04.320 | 00:00    0ms    0M    0G  12/ 15U [Build] Att at 60
2024-11-14 11:35:04.322 | ERROR    | sc2.main:initialize_first_step:140 - Caught unknown exception in AI on_start: UpgradeId.TERRANVEHICLEANDSHIPWEAPONSLEVEL1
Traceback (most recent call last):

  File "c:\Users\Lena\AppData\Local\Bots\DemonT\run.py", line 18, in <module>
    main()
    └ <function main at 0x00000164CA88F600>

  File "c:\Users\Lena\AppData\Local\Bots\DemonT\run.py", line 13, in main
    result, opponentid = run_ladder_game(protoss_bot)
                         │               └ <sc2.player.Bot object at 0x00000164CA8BA9D0>
                         └ <function run_ladder_game at 0x00000164AF17E5C0>

  File "c:\Users\Lena\AppData\Local\Bots\DemonT\ladder.py", line 39, in run_ladder_game
    return stand_alone_game(bot), None
           │                └ <sc2.player.Bot object at 0x00000164CA8BA9D0>
           └ <function stand_alone_game at 0x00000164C7FE49A0>

  File "c:\Users\Lena\AppData\Local\Bots\DemonT\ladder.py", line 166, in stand_alone_game
    return sc2.main.run_game(sc2.maps.get(map_name), [bot, Computer(Race.Zerg, Difficulty.VeryHard)], realtime=False,)
           │   │    │        │   │    │   │           │    │        │    │     │          └ <Difficulty.VeryHard: 7>
           │   │    │        │   │    │   │           │    │        │    │     └ <enum 'Difficulty'>
           │   │    │        │   │    │   │           │    │        │    └ <Race.Zerg: 2>
           │   │    │        │   │    │   │           │    │        └ <enum 'Race'>
           │   │    │        │   │    │   │           │    └ <class 'sc2.player.Computer'>
           │   │    │        │   │    │   │           └ <sc2.player.Bot object at 0x00000164CA8BA9D0>
           │   │    │        │   │    │   └ 'Equilibrium513AIE'
           │   │    │        │   │    └ <function get at 0x00000164C804D8A0>
           │   │    │        │   └ <module 'sc2.maps' from 'c:\\Users\\Lena\\AppData\\Local\\Bots\\DemonT\\sc2\\maps.py'>
           │   │    │        └ <module 'sc2' from 'c:\\Users\\Lena\\AppData\\Local\\Bots\\DemonT\\sc2\\__init__.py'>
           │   │    └ <function run_game at 0x00000164C817DE40>
           │   └ <module 'sc2.main' from 'c:\\Users\\Lena\\AppData\\Local\\Bots\\DemonT\\sc2\\main.py'>
           └ <module 'sc2' from 'c:\\Users\\Lena\\AppData\\Local\\Bots\\DemonT\\sc2\\__init__.py'>

  File "c:\Users\Lena\AppData\Local\Bots\DemonT\sc2\main.py", line 496, in run_game
    result: Result = asyncio.run(_host_game(map_settings, players, **kwargs))
                     │       │   │          │             │          └ {'realtime': False}
                     │       │   │          │             └ [<sc2.player.Bot object at 0x00000164CA8BA9D0>, <sc2.player.Computer object at 0x00000164C9494850>]
                     │       │   │          └ Map(D:\StarCraft II\maps\Equilibrium513AIE.SC2Map)
                     │       │   └ <function _host_game at 0x00000164C817D9E0>
                     │       └ <function run at 0x00000164B0DF60C0>
                     └ <module 'asyncio' from 'C:\\Users\\Lena\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\asyncio\\__init__.py'>

  File "C:\Users\Lena\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 190, in run
    return runner.run(main)
           │      │   └ <coroutine object _host_game at 0x00000164CA814F60>
           │      └ <function Runner.run at 0x00000164B0E760C0>
           └ <asyncio.runners.Runner object at 0x00000164C94FE790>
  File "C:\Users\Lena\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           │    │     │                  └ <Task pending name='Task-1' coro=<_host_game() running at c:\Users\Lena\AppData\Local\Bots\DemonT\sc2\main.py:369> cb=[_run_u...
           │    │     └ <function BaseEventLoop.run_until_complete at 0x00000164B0E6FBA0>
           │    └ <ProactorEventLoop running=True closed=False debug=False>
           └ <asyncio.runners.Runner object at 0x00000164C94FE790>
  File "C:\Users\Lena\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 640, in run_until_complete
    self.run_forever()
    │    └ <function ProactorEventLoop.run_forever at 0x00000164B136FBA0>
    └ <ProactorEventLoop running=True closed=False debug=False>
  File "C:\Users\Lena\AppData\Local\Programs\Python\Python311\Lib\asyncio\windows_events.py", line 321, in run_forever
    super().run_forever()
  File "C:\Users\Lena\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 607, in run_forever
    self._run_once()
    │    └ <function BaseEventLoop._run_once at 0x00000164B0E75940>
    └ <ProactorEventLoop running=True closed=False debug=False>
  File "C:\Users\Lena\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 1922, in _run_once
    handle._run()
    │      └ <function Handle._run at 0x00000164B0DBAF20>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>
  File "C:\Users\Lena\AppData\Local\Programs\Python\Python311\Lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
    │    │            │    │           │    └ <member '_args' of 'Handle' objects>
    │    │            │    │           └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    │            │    └ <member '_callback' of 'Handle' objects>
    │    │            └ <Handle Task.task_wakeup(<Future finished result=None>)>
    │    └ <member '_context' of 'Handle' objects>
    └ <Handle Task.task_wakeup(<Future finished result=None>)>

  File "c:\Users\Lena\AppData\Local\Bots\DemonT\sc2\main.py", line 369, in _host_game
    result = await _play_game(players[0], client, realtime, portconfig, game_time_limit, rgb_render_config)
                   │          │           │       │         │           │                └ None
                   │          │           │       │         │           └ None
                   │          │           │       │         └ None
                   │          │           │       └ False
                   │          │           └ <sc2.client.Client object at 0x00000164C934DCD0>
                   │          └ [<sc2.player.Bot object at 0x00000164CA8BA9D0>, <sc2.player.Computer object at 0x00000164C9494850>]
                   └ <function _play_game at 0x00000164C817D800>

  File "c:\Users\Lena\AppData\Local\Bots\DemonT\sc2\main.py", line 226, in _play_game
    result = await _play_game_ai(client, player_id, player.ai, realtime, game_time_limit)
                   │             │       │          │      │   │         └ None
                   │             │       │          │      │   └ False
                   │             │       │          │      └ <dummy.dummy.LadderBot object at 0x00000164AEE9DF50>
                   │             │       │          └ <sc2.player.Bot object at 0x00000164CA8BA9D0>
                   │             │       └ 1
                   │             └ <sc2.client.Client object at 0x00000164C934DCD0>
                   └ <function _play_game_ai at 0x00000164C817D6C0>

  File "c:\Users\Lena\AppData\Local\Bots\DemonT\sc2\main.py", line 145, in _play_game_ai
    result = await initialize_first_step()
                   └ <function _play_game_ai.<locals>.initialize_first_step at 0x00000164CA8D4180>

> File "c:\Users\Lena\AppData\Local\Bots\DemonT\sc2\main.py", line 136, in initialize_first_step
    await ai.on_start()
          │  └ <function KnowledgeBot.on_start at 0x00000164C82EACA0>
          └ <dummy.dummy.LadderBot object at 0x00000164AEE9DF50>

  File "c:\Users\Lena\AppData\Local\Bots\DemonT\sharpy\knowledges\knowledge_bot.py", line 71, in on_start
    await self.knowledge.start()
          │    │         └ <function Knowledge.start at 0x00000164C829B060>
          │    └ <sharpy.knowledges.knowledge.Knowledge object at 0x00000164AEEEB710>
          └ <dummy.dummy.LadderBot object at 0x00000164AEE9DF50>

  File "c:\Users\Lena\AppData\Local\Bots\DemonT\sharpy\knowledges\knowledge.py", line 151, in start
    await manager.post_start()
          │       └ <function ActManager.post_start at 0x00000164C802DBC0>
          └ <sharpy.managers.core.act_manager.ActManager object at 0x00000164CA9427D0>

  File "c:\Users\Lena\AppData\Local\Bots\DemonT\sharpy\managers\core\act_manager.py", line 25, in post_start
    self._act = await self._act_or_func()
    │                 │    └ <bound method Banshees.create_plan of <dummy.dummy.LadderBot object at 0x00000164AEE9DF50>>
    │                 └ <sharpy.managers.core.act_manager.ActManager object at 0x00000164CA9427D0>
    └ <sharpy.managers.core.act_manager.ActManager object at 0x00000164CA9427D0>

  File "c:\Users\Lena\AppData\Local\Bots\DemonT\dummy\dummy.py", line 123, in create_plan
    Tech(UpgradeId.TERRANVEHICLEANDSHIPWEAPONSLEVEL1),
    │    │         └ UpgradeId.TERRANVEHICLEANDSHIPWEAPONSLEVEL1
    │    └ <enum 'UpgradeId'>
    └ <class 'sharpy.plans.acts.tech.Tech'>

  File "c:\Users\Lena\AppData\Local\Bots\DemonT\sharpy\plans\acts\tech.py", line 39, in __init__
    from_building = UPGRADE_RESEARCHED_FROM[self.upgrade_type]
                    │                       │    └ UpgradeId.TERRANVEHICLEANDSHIPWEAPONSLEVEL1
                    │                       └ <sharpy.plans.acts.tech.Tech object at 0x00000164CB332A50>
                    └ {UpgradeId.ADEPTPIERCINGATTACK: UnitTypeId.TWILIGHTCOUNCIL, UpgradeId.ANABOLICSYNTHESIS: UnitTypeId.ULTRALISKCAVERN, UpgradeI...

KeyError: UpgradeId.TERRANVEHICLEANDSHIPWEAPONSLEVEL1
2024-11-14 11:35:04.394 | ERROR    | sc2.main:initialize_first_step:141 - Resigning due to previous error
2024-11-14 11:35:04.395 | Result: Defeat
2024-11-14 11:35:04.396 | Duration: 00:00
2024-11-14 11:35:04.397 | Step time avg: 0
2024-11-14 11:35:04.397 | Step time max: 0
2024-11-14 11:35:04.398 | [LostUnitsContents] My lost units minerals and gas: (0, 0)
2024-11-14 11:35:04.398 | [LostUnitsContents] My lost units minerals by score: 0.0
2024-11-14 11:35:04.399 | [LostUnitsContents] Enemy lost units minerals and gas: (0, 0)
2024-11-14 11:35:04.400 | [LostUnitsContents] Enemy lost units minerals by score: 0.0
2024-11-14 11:35:04.401 | [GameAnalyzerEnd] Own units:
2024-11-14 11:35:04.401 | [GameAnalyzerEnd] Enemy units:
2024-11-14 11:35:04.401 | [GameAnalyzerEnd] Minerals max 0 Average 0
2024-11-14 11:35:04.402 | [GameAnalyzerEnd] Vespene max 0 Average 0
2024-11-14 11:35:04.403 | INFO     | sc2.main:_play_game:228 - Result for player 1 - Bot LadderBot(Terran): Defeat
2024-11-14 11:35:05.751 | INFO     | sc2.protocol:_execute:72 - Client status changed to Status.launched (was Status.in_game)
2024-11-14 11:35:05.766 | INFO     | sc2.protocol:_execute:72 - Client status changed to Status.quit (was Status.launched)
2024-11-14 11:35:05.767 | INFO     | sc2.sc2process:_close_connection:231 - Closing connection at 52094...
2024-11-14 11:35:05.769 | INFO     | sc2.sc2process:kill_all:33 - kill_switch: Process cleanup for 1 processes
